Regression/Checksum/backend/amrex_backend.py

```diff
@@ -0,0 +1,81 @@
+"""
+Copyright 2021
+
+This file is part of WarpX
+
+License: BSD-3-Clause-LBNL
+"""
+
+import re
+import numpy as np
+import yt
+from yt.frontends.boxlib.data_structures import AMReXDataset
+
+yt.funcs.mylog.setLevel(50)
+
+class Backend:
+    ''' Use AMReXDataset as the backend reader to read AMReX plotfiles
+    '''
+
+    def __init__(self, filename):
+        ''' Constructor: store the dataset object
+        '''
+
+        self.dataset = AMReXDataset(filename)
+
+    def fields_list(self):
+        ''' Return the list of fields defined on the grid
+        '''
+
+        return [item for item in self.dataset.field_list if item[0] == 'boxlib']
+
+    def species_list(self):
+        ''' Return the list of species in the dataset
+        '''
+
+        return set([item[0] for item in self.dataset.field_list if
+                            item[1][:9] == 'particle_' and item[0] != 'all'])
+
+    def n_levels(self):
+        ''' Return the number of MR levels in the dataset
+        '''
+
+        return self.dataset.max_level+1;
+
+    def get_field_checksum(self, lev, field, test_name):
+        ''' Calculate the checksum for a given field at a given level in the dataset
+        '''
+
+        # Boolean variable parsed from test name needed for workaround below
+        galilean = True if re.search('galilean', test_name) else False
+        print("+++++ {} ++++ {} +++++".format(test_name, galilean))
+
+        lev_grids = [grid for grid in self.dataset.index.grids if grid.Level == lev]
+        if not (galilean):
+            # Warning: For now, we assume all levels are rectangular
+            LeftEdge = np.min(
+                np.array([grid.LeftEdge.v for grid in lev_grids]), axis=0)
+            all_data_level = self.dataset.covering_grid(
+                level=lev, left_edge=LeftEdge, dims=self.dataset.domain_dimensions)
+            Q = all_data_level[field].v.squeeze()
+            return np.sum(np.abs(Q))
+        # Workaround for Galilean tests: the standard procedure above
+        # does not seem to read 2D fields data correctly
+        elif (galilean):
+            Q = self.dataset.index.grids[lev][field].v.squeeze()
+            return np.sum(np.abs(Q))
```

Comment:
@ax3l I think I removed this workaround in #1536, about two months ago. I think we should not (re-)introduce a difference between Galilean and non-Galilean here. This might explain why something weird is happening with the test `galilean_hybrid_2d` (not sure why you don't see other problems with other Galilean tests, though, but one problem at at time...).

In other words, I think you might have moved this piece of code here (with the dichotomy `if not (galilean) ... elif (galilean)`) from a version of checksum.py that was older than #1536, where it was modified. Then, while rebasing this PR on development, it is possible that git missed the conflict because the same piece of code was moved here into a different file, so git would not catch that there was a conflict in the changes. Does this make sense to you?

Reply:
Ah very good catch, thanks @EZoni. The files in this PR come from Hipace++, and were copied from WarpX months ago, so I think you're totally right on what happened. Let's just include your changes from #1536 here, and then I'll also do them in Hipace++. The checksum module is much cleaner without this special Galilean handling indeed.